iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractVoltageLevel.java

```diff
@@ -123,41 +124,45 @@ public VoltageLevel setHighVoltageLimit(double highVoltageLimit) {
     }
 
     @Override
-    public <T extends Connectable> FluentIterable<T> getConnectables(Class<T> clazz) {
+    public <T extends Connectable> Iterable<T> getConnectables(Class<T> clazz) {
         Iterable<Terminal> terminals = getTerminals();
         return FluentIterable.from(terminals)
                 .transform(Terminal::getConnectable)
-                .filter(clazz);
+                .filter(clazz)
+                .toSet();
     }
 
     @Override
     public <T extends Connectable> Stream<T> getConnectableStream(Class<T> clazz) {
         return getTerminalStream()
                 .map(Terminal::getConnectable)
                 .filter(clazz::isInstance)
-                .map(clazz::cast);
+                .map(clazz::cast)
+                .distinct();
     }
 
     @Override
     public <T extends Connectable> int getConnectableCount(Class<T> clazz) {
-        return getConnectables(clazz).size();
+        return Ints.checkedCast(getConnectableStream(clazz).count());
```

Comment:
Do I get it right that using count() is a better practice than using size(), and that count() returns a long so you need to cast it to an int?

Reply:
No, I only used `count()` because `size()` does not exist for streams or for iterables (only for fluentiterables)